recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -82,19 +82,23 @@ this.NormandyApi = {
     return this.absolutify(url);
   },
 
-  async fetchRecipes(filters = {enabled: true}) {
-    const signedRecipesUrl = await this.getApiUrl("recipe-signed");
-    const recipesResponse = await this.get(signedRecipesUrl, filters);
-    const rawText = await recipesResponse.text();
-    const recipesWithSigs = JSON.parse(rawText);
+  async fetchSignedObjects(type, filters) {
+    const signedObjectsUrl = await this.getApiUrl(`${type}-signed`);
+    const objectsResponse = await this.get(signedObjectsUrl, filters);
+    const rawText = await objectsResponse.text();
+    const objectsWithSigs = JSON.parse(rawText);
+
+    const verifiedObjects = [];
 
-    const verifiedRecipes = [];
+    for (const objectWithSig of objectsWithSigs) {
+      const {signature, x5u} = objectWithSig.signature;
+      const object = objectWithSig[type];
 
-    for (const {recipe, signature: {signature, x5u}} of recipesWithSigs) {
-      const serialized = CanonicalJSON.stringify(recipe);
+      const serialized = CanonicalJSON.stringify(object);
       if (!rawText.includes(serialized)) {
```

Comment:
We could leave this check out entirely, and it would be fine because it still wouldn't match the signatures. This is mostly a benefit to give better error messages to developers, especially in tests and QA.

The reason it is a substring check is because `rawText` includes both the object that is signed and the signature for that object. It isn't required that the entire respons be in canonical json format, only the object that is signed, which is what `serialized` is a representation of.

I'll add a comment.